build-system/server/lazy-build.js

```diff
@@ -63,8 +68,16 @@ async function lazyBuild(url, matcher, bundles, buildFunc, next) {
   if (match && match.length == 2) {
     const name = maybeGetUnminifiedName(bundles, match[1]);
     const bundle = bundles[name];
-    if (bundle) {
+
+    if (bundle && argv.compiled) {
       await build(bundles, name, buildFunc);
+    } else if (bundle && !esbuildWatchedExtensions.has(name)) {
```

Comment:
On re-reading this PR, would it be possible to encapsulate this logic inside `build()`? `lazyBuild` merely cares about whether or not to build a given target when it is fetched from the server. The business logic about how the target is actually built ought to live inside `build()` and not here.